drivers/common/virtio/virtio_pci_net.c

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2022 NVIDIA Corporation & Affiliates
+ */
+#ifdef RTE_EXEC_ENV_LINUX
+ #include <dirent.h>
+ #include <fcntl.h>
+#endif
+
+#include <rte_io.h>
+#include <rte_bus.h>
+
+#include "virtio_pci.h"
+#include "virtio_logs.h"
+#include "virtqueue.h"
+
+static uint16_t
+modern_net_get_queue_num(struct virtio_hw *hw)
```

Comment:
cut down this routine. it was overkill from the beginning.

struct virtio_pci_common_cfg {} structure has num_queues field for all the device types including net and blk.
This needs to reflect the maximum queue value regardless of device type.

Use it uniformly for net and blk.
and put in common code.

Reply:
num_queue include control queue and admin queue.
VDPA　driver use this to get how many IO queue device can support.
so, shoule only return number of IO queue.
So, blk and net is different also.
we should keep this code , but only remove admin queue and controler queue logic.

Reply:
@kailiangz1 the logic should be in get_vqs_num() callbkc to be,

get_hw_num_vqs() -1 for net cvq.

Reply:
It should not like this, get_hw_num_vqs to get from common config's num_queues is hard to judge
whether the number include cvq or admin vq.
Directly get from virtio_net_config obey the spec and do need to calculate.

Reply:
@kailiangz1 num_vqs is generic for all the protocols regardless of device type. This is the virtio pci layer provides.
vdpa net and block layer can derive its own value from low layer.

This is simple math. no need to judge/guess anything.

Reply:
we need to judge whether cvq is enabled and whether admin vq is enabled.